python/tank/commands/core_localize.py

```diff
@@ -78,114 +78,90 @@ def run_interactive(self, log, args):
-    """ 
+    """
 
     pc = pipelineconfig_factory.from_path(pc_root_path)
 
     log.info("")
     if pc.is_localized():
         raise TankError("Looks like your current pipeline configuration already has a local install of the core!")
-    
+
     core_api_root = pc.get_install_location()
-    
-    log.info("This will copy the Core API in %s into the Pipeline configuration %s." % (core_api_root, pc_root_path))
+
+    log.info("This will copy the Core API in %s \ninto the Pipeline configuration %s." % (core_api_root, pc_root_path))
     log.info("")
     if not suppress_prompts:
         # check with user if they wanna continue
         if not console_utils.ask_yn_question("Do you want to proceed"):
             # user says no!
             log.info("Operation cancelled.")
             return
-    
+
     # proceed with setup
     log.info("")
-    
-    # first get a list of all bundle descriptors
-    # key by descriptor repr, which ensures uniqueness   
-    # at this point we also store the path to each descriptor
-    # before we make any changes to any config files 
-    descriptors = {}
-    for env_name in pc.get_environments():
-        
-        env_obj = pc.get_environment(env_name)
-        
-        for engine in env_obj.get_engines():
-            descriptor = env_obj.get_engine_descriptor(engine)
-            descriptors[ repr(descriptor) ] = { "name": str(descriptor), "path": descriptor.get_path() }
-            
-            for app in env_obj.get_apps(engine):
-                descriptor = env_obj.get_app_descriptor(engine, app)
-                descriptors[ repr(descriptor) ] = { "name": str(descriptor), "path": descriptor.get_path() }
-                
-        for framework in env_obj.get_frameworks():
-            descriptor = env_obj.get_framework_descriptor(framework)
-            descriptors[ repr(descriptor) ] = { "name": str(descriptor), "path": descriptor.get_path() }
-    
-    
-    source_core = os.path.join(core_api_root, "install", "core")
-    target_core = os.path.join(pc_root_path, "install", "core")
-    backup_location = os.path.join(pc_root_path, "install", "core.backup")
-        
-    old_umask = os.umask(0)
-    try:        
 
-        # step 1. copy this into backup location
+    try:
+
+        ######################################################################
+        #
+        # step one - copy all the contents of the install location across except
+        # for the contents in the core and core.backup folders - these are handled
+        # explicitly
+
+        source_install_path = os.path.join(core_api_root, "install")
+        target_install_path = os.path.join(pc_root_path, "install")
+
+        names = os.listdir(source_install_path)
+        for name in names:
+
+            if name in ["core", "core.backup"]:
+                # skip now and handle separately
+                continue
+
+            if name.startswith(".") or name.startswith("_"):
```

Comment:
Makes sense. Just continuing your thought, maybe we should have a constant in core that is a list of files we always ignore when copying/moving. Then we can be specific in case we do want to have `.` or `_` filed/dirs at some point down the line.